### DISCUSSION TITLE
Is it possible to use IS instead of equal sign for boolean conditions

### DISCUSSION THREAD

**Discussion by damirabdul** [jOOQ/jOOQ#14167] [Q&A] [answered]
Hello!

I'm using Postgres database with default read-commited isolation level and JOOQ 3.16.6. Let's say I have table `user` which was created as
```
CREATE TABLE user
(
    id         bigserial primary key,
    name       varchar,
    is_deleted boolean default false
);

CREATE UNIQUE INDEX IF NOT EXISTS user_name_unique ON user (name) WHERE user.is_deleted IS FALSE;
```
Please notice, that I have a partial unique index in this table.

### Problem

There are 2 transactions which are trying  to create user with the same name

`INSERT INTO user (name) VALUES ('John') ON CONFLICT (name) WHERE user.is_deleted IS FALSE DO UPDATE SET id = EXCLUDED.id RETURNING *;`

With this query I want to achieve, that first transaction will be committed successfully and the second one won't fail, but return rows which is was trying to insert.

The problem is, that I can't build this query with JOOQ, since JOOQ is generating `WHERE user.is_deleted = FALSE` instead of `WHERE user.is_deleted IS FALSE` and query fails with error **ERROR: there is no unique or exclusion constraint matching the ON CONFLICT specification**.

Is there any way to generate condition with IS keyword instead of equal sign? 

Thanks in advance!

**Comment by lukaseder** [comment, marked as answer]
Thanks for your message.

There's a feature request to support this syntax, see here: https://github.com/jOOQ/jOOQ/issues/4153. It's very rarely useful and conflicts with an existing feature, see the issue description.

For the time being, just use a plain SQL template, as always when some syntax isn't available through jOOQ:
https://www.jooq.org/doc/latest/manual/sql-building/plain-sql-templating

**Reply by damirabdul** [reply]
Thanks for a quick answer, appreciate!
